pkg/catalog/catalog.go

```diff
@@ -21,6 +22,25 @@ var DevfileRegistries = []string{
 	"https://che-devfile-registry.openshift.io/",
 }
 
+var devfileIndicesMutex sync.Mutex
+
+// getDevfileIndexWith populates the given entries array with DevfileIndexEntry found in the given registry link.
+// This function is meant to be called concurrently by go routines so that all registries can be processed at the
+// same time, using a WaitGroup to make sure we get the results from all the registries before further processing.
+// If an error occurs, it will be populated using the given error parameter.
+func getDevfileIndexWith(link string, entries *[]DevfileIndexEntry, err error, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	devfileIndexLink := link + "/devfiles/index.json"
+	indexEntries, err := GetDevfileIndex(devfileIndexLink)
```

Comment:
It seems we are not checking the error https://travis-ci.com/github/openshift/odo/jobs/329269812#L271